src/components/Search/SearchProvider.tsx

```diff
@@ -21,35 +22,90 @@ export type SearchProviderProps = {
   navigate: (url: string) => void;
   children: ReactNode | ((props: SearchProviderChildrenProps) => ReactNode);
   detailsId?: string;
+  initialData?: OpenLibraryResponse;
+  initialQuery?: string;
+  initialError?: string | null;
 };
 
+const transformOpenLibraryBook = (doc: OpenLibraryBook): Book => ({
+  key: doc.key,
+  title: doc.title || '',
+  author_name: doc.author_name ?? [],
+  first_publish_year: doc.first_publish_year,
+  cover_i: doc.cover_i,
+  subject: doc.subject ?? [],
+  isbn: doc.isbn ?? [],
+  publisher: doc.publisher ?? [],
+});
+
 export const SearchProvider = ({
   children,
   currentPage,
   navigate,
+  initialData,
+  initialQuery,
+  initialError,
 }: SearchProviderProps) => {
-  const initialSearchValue =
-    localStorage.getItem(STORAGE_KEYS.PREV_SEARCH_VALUE) || '';
+  const [searchValue, setSearchValueState] = useState(initialQuery || '');
+  const [savedSearchTerm, setSavedSearchTerm] = useState(initialQuery || '');
+  const [hasInitialized, setHasInitialized] = useState(false);
+  const [shouldUseInitialData, setShouldUseInitialData] =
+    useState(!!initialData);
+
+  useEffect(() => {
+    if (typeof window !== 'undefined' && !hasInitialized) {
```

Comment:
```suggestion
    if (!hasInitialized) {
```
когда в файле есть 'use client', компонент рендерится только на клиенте и эта проверка избыточна